pkg/test/istio.io/tasks/security/mtls-migration/mtls-migration_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+package tests
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/istio.io/examples"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+
+	"istio.io/istio/pkg/test/framework/components/istio"
+)
+
+var (
+	ist istio.Instance
+)
+
+func TestMain(m *testing.M) {
+	framework.NewSuite("mtls-migration", m).
+		SetupOnEnv(environment.Kube, istio.Setup(&ist, nil)).
+		RequireEnvironment(environment.Kube).
+		Run()
+}
+
+//https://istio.io/docs/tasks/security/mtls-migration/
+//https://github.com/istio/istio.io/blob/release-1.2/content/docs/tasks/security/mtls-migration/index.md
+func TestMTLS(t *testing.T) {
+	ex := examples.New(t, "mtls-migration")
+
+	//The following line is just an example of how to use addfile.
+	ex.AddFile("istio-system", "samples/sleep/sleep.yaml")
```

Comment:
Nit - we can have these functions return `ex` as well, so you could do a more fluent

```
ex.
  AddFile().
  AddScript().
  Run()
```